### ISSUE TITLE
Stuck at "Starting Tectonic" : on master node, etcd does not start

### ISSUE THREAD

**Issue by jdall** [coreos/tectonic-installer#2965]
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

### Tectonic version ([release](https://github.com/coreos/tectonic-installer/releases) or commit hash):
```
Downloaded tectonic_1.8.4-tectonic.3.zip, but the logs state tectonic_1.8.4-tectonic.2, so which one is it?
```

### Terraform version (`terraform version`):
```
Terraform v0.10.7

Your version of Terraform is out of date! The latest version
is 0.11.3. You can update by downloading from www.terraform.io/downloads.html
```
### Platform (aws|azure|openstack|metal|vmware):

```
metal
```

### What happened?
```
Selected a GUI install on bare-metal. Tectonic Installer stuck at "Starting Tectonic". 

On the master node, etcd is not starting. The log show (timestamps removed for shorter lines):

etcd-wrapper[31374]: 2018-02-16 07:10:15.315731 I | pkg/flags: recognized and used environment variable ETCD_ADVERTISE_CLIENT_URLS=https://master1.k8s.eurisco:2379
etcd-wrapper[31374]: 2018-02-16 07:10:15.322214 I | pkg/flags: recognized and used environment variable ETCD_CERT_FILE=/etc/ssl/certs/etcd/server.crt
etcd-wrapper[31374]: 2018-02-16 07:10:15.322272 I | pkg/flags: recognized and used environment variable ETCD_CLIENT_CERT_AUTH=true
etcd-wrapper[31374]: 2018-02-16 07:10:15.322334 I | pkg/flags: recognized and used environment variable ETCD_DATA_DIR=/var/lib/etcd
etcd-wrapper[31374]: 2018-02-16 07:10:15.322399 I | pkg/flags: recognized and used environment variable ETCD_INITIAL_ADVERTISE_PEER_URLS=https://master1.k8s.eurisco:2380
etcd-wrapper[31374]: 2018-02-16 07:10:15.322460 I | pkg/flags: recognized and used environment variable ETCD_INITIAL_CLUSTER=master1=https://master1.k8s.eurisco:2380
etcd-wrapper[31374]: 2018-02-16 07:10:15.322514 I | pkg/flags: recognized and used environment variable ETCD_KEY_FILE=/etc/ssl/certs/etcd/server.key
etcd-wrapper[31374]: 2018-02-16 07:10:15.322573 I | pkg/flags: recognized and used environment variable ETCD_LISTEN_CLIENT_URLS=https://0.0.0.0:2379
etcd-wrapper[31374]: 2018-02-16 07:10:15.322641 I | pkg/flags: recognized and used environment variable ETCD_LISTEN_PEER_URLS=https://0.0.0.0:2380
etcd-wrapper[31374]: 2018-02-16 07:10:15.322710 I | pkg/flags: recognized and used environment variable ETCD_NAME=master
etcd-wrapper[31374]: 2018-02-16 07:10:15.322766 I | pkg/flags: recognized and used environment variable ETCD_PEER_CERT_FILE=/etc/ssl/certs/etcd/peer.crt
etcd-wrapper[31374]: 2018-02-16 07:10:15.322821 I | pkg/flags: recognized and used environment variable ETCD_PEER_CLIENT_CERT_AUTH=true
etcd-wrapper[31374]: 2018-02-16 07:10:15.322875 I | pkg/flags: recognized and used environment variable ETCD_PEER_KEY_FILE=/etc/ssl/certs/etcd/peer.key
etcd-wrapper[31374]: 2018-02-16 07:10:15.322931 I | pkg/flags: recognized and used environment variable ETCD_PEER_TRUSTED_CA_FILE=/etc/ssl/certs/etcd/peer-ca.crt
etcd-wrapper[31374]: 2018-02-16 07:10:15.322986 I | pkg/flags: recognized and used environment variable ETCD_STRICT_RECONFIG_CHECK=true
etcd-wrapper[31374]: 2018-02-16 07:10:15.323033 I | pkg/flags: recognized and used environment variable ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd/server-ca.crt
etcd-wrapper[31374]: 2018-02-16 07:10:15.323089 W | pkg/flags: unrecognized environment variable ETCD_SSL_DIR=/etc/ssl/etcd
etcd-wrapper[31374]: 2018-02-16 07:10:15.323138 W | pkg/flags: unrecognized environment variable ETCD_USER=etcd
etcd-wrapper[31374]: 2018-02-16 07:10:15.323185 W | pkg/flags: unrecognized environment variable ETCD_IMAGE_TAG=v3.2.0
etcd-wrapper[31374]: 2018-02-16 07:10:15.323244 I | etcdmain: etcd Version: 3.2.0
etcd-wrapper[31374]: 2018-02-16 07:10:15.323288 I | etcdmain: Git SHA: 66722b1
etcd-wrapper[31374]: 2018-02-16 07:10:15.323335 I | etcdmain: Go Version: go1.8.3
etcd-wrapper[31374]: 2018-02-16 07:10:15.323376 I | etcdmain: Go OS/Arch: linux/amd64
etcd-wrapper[31374]: 2018-02-16 07:10:15.323421 I | etcdmain: setting maximum number of CPUs to 1, total number of available CPUs is 1
etcd-wrapper[31374]: 2018-02-16 07:10:15.331190 I | embed: peerTLS: cert = /etc/ssl/certs/etcd/peer.crt, key = /etc/ssl/certs/etcd/peer.key, ca = , trusted-ca = /etc/ssl/certs/etcd/peer-ca.crt, client-cert-auth = true
etcd-wrapper[31374]: 2018-02-16 07:10:15.331311 C | etcdmain: open /etc/ssl/certs/etcd/peer.crt: no such file or directory
systemd[1]: etcd-member.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: etcd-member.service: Failed with result 'exit-code'.
systemd[1]: Failed to start etcd (System Application Container).
```

### What you expected to happen?
```
etcd should have started and Tectonic Installer should have reached the "Installation Complete" step
```

### How to reproduce it (as minimally and precisely as possible)?
```
Basically followed the guidelines in the https://coreos.com/tectonic/docs/latest/install/bare-metal/index.html#4-tectonic-installer
```

### Anything else we need to know?
```
Running matchbox and dnsmasq as services on a separate Container Linux system (matchbox started using the matchbox-for-tectonic.service and dnsmasq running as in proxy-dhcp mode). 

Initially, files from '~/matchbox-master/examples/groups/bootkube-install/' were copied to '/var/lib/matchbox/groups' and renamed as node1.json->master1.json, node2.json->worker1.json, node3.json->worker2.json. Finally, they were modified to match the hardware.

After running the linux Tectonic Installer, the /var/lib/matchbox/groups also contains coreos-install-{master1,worker1,worker2}.json and eurisco-staging-{master1,worker1,worker2}.json

The eurisco-staging-master1.json (attached) has no reference to e.g. the '/etc/ssl/certs/etcd/peer-ca.crt' file, so where is the that setting coming from? Is it the etcd install that provides it?

```
[eurisco-staging-master1.json](https://github.com/coreos/tectonic-installer/files/1730663/eurisco-staging-master1.json.txt)

### References
```
None.
```

### Environment
```
Matchbox is running on a VirtualBox VM on a Debian 9 linux host system.

Nodes are running on Proxmox VMs.

The Tectonic Installer is executed from the Debian 9 linux system.
```
### Desired Feature
```
No feature requested, other than that the installer provides valid configurations for the nodes.
```

### Other Information
```
None.
```

**Comment by squat** [comment]
@jdall the etcd TLS assets are created by Terraform at runtime and scp’d onto all master nodes by this Terraform resource [0]. Judging from your description I suspect that your installation never progressed beyond the “creating null.resource.etcd_secrets” step. Can you provide logs of the Terraform output to confirm this? The most likely culprit is that the host that is running the Tectonic Installer was unable to establish a SSH connection to the master nodes. This could be because for example the provisioned SSH key cannot be found by the `ssh` binary, or because DNS is misconfigured so the nodes cannot be reached using their domain name. Please verify that you are able to ssh onto the master nodes from the host running the Tectonic Installer the domain name configured for that node in the installer.

[0] https://github.com/coreos/tectonic-installer/blob/1.8.4-tectonic.3/platforms/metal/remote.tf#L1

**Comment by jdall** [comment]
Thanks for the prompt feedback, squat.

It took some effort, but finally I got beyond the "creating null.resource.etcd_secrets" step. 

I can connect to the nodes using SSH and the installer itself has also established communication with the nodes, because some of the details I entered in the installer are present on them.

The log file I've attached reflects a cluster reset, I tried in the installer, when it had not moved beyond the "Starting Tectonic" for quite a while. Before rebooting the nodes, I used the "dd if=/dev/zero of=/dev/sda count=1" trick do erradicate the boot record.

I wonder if I should not have copied the group files from the matchbox examples initially?

[22912.log](https://github.com/coreos/tectonic-installer/files/1730783/22912.log.txt)

**Comment by jdall** [comment]
Update: I removed all files from /var/lib/matchbox/groups and performed a cluster reset. That solved the crt issue.

After a while (5+ minutes) I also got past the "Starting Tectonic" step (nice progress bar), and the step completed. The rest was eacy-peasy.

I've got a nice Tectonic Console now ;)

Thanks again, squid, for the prompt feedback, that made me re-thing the whole process, and finally figuring out the issue.
